### PR TITLE
Float APIs - Part1

### LINKED ISSUE #484: VM Float API Implementation
Referenced from the Ruby Ver. 2.4.1. Float data type API:

- [ ] Constant `Float::INFINITE` stands for the number is infinite
- [ ] Constant `Float::NAN` stands for "Not a Number"

- [ ] `Float#modulo` Synonym to the `%` operator (Get the quotient from the divide operation)
- [ ] `Float#quo` Synonym to the `/` operator (Get the quotient from the divide operation)
- [ ] `Float#abs` Return positive number of the float. "Formally as the distance between 0 to the number of the float in 1-dimensional axis ..."
- [ ] `Float#angle` (and `Float#arg`) If positive -> return 0; else -> return PI
- [ ] `Float#ceil` Returns smallest number greater than or equal to the receiver
- [ ] `Float#floor` Returns largest number smaller than or equal to the receiver
- [ ] `Float#round` Rounds float to a given precision in decimal digits (default 0 digits). Precision may be negative. Returns a floating point number when `n` digits is more than zero.
- [ ] `Float#finite?` Returns `true` if receiver isn't  equal to `Float::INFINITE`
- [ ] `Float#infinite?` Returns `true` if receiver is equal to `Float::INFINITE`
- [ ] `Float#divmod` Returns array which is in format of [ (Integer) Quotient, (Float) Remainder ]
- [ ] `Float#nan?` Returns `true` if receiver is equal to `Float::NAN`
- [ ] `Float#positive?` Returns `true` if float number is larger than `0.0`
- [ ] `Float#negative?` Returns `true` if float number is less than `0.0`
- [ ] `Float#zero?` Returns `true` if float number is equal to `0.0`
- [ ] `Float#rationalize` Convert to something called `Rational` data type (Currently not supported but from my point of view, it is worth being addressed in this issue)

### PR DESCRIPTION
**Pull request by st0012**
This PR is for continuing the work at #615 (thanks @SD10 for previous works)

It implements:
- [x] `Float#abs` Return positive number of the float. "Formally as the distance between 0 to the number of the float in 1-dimensional axis ..."
- [x] `Float#ceil` Returns the smallest number greater than or equal to the receiver
- [x] `Float#floor` Returns the largest number smaller than or equal to the receiver
- [x] `Float#round` Rounds float to a given precision in decimal digits (default 0 digits). Precision may be negative. Returns a floating-point number when `n` digits are more than zero.

- [x] `Float#positive?` Returns `true` if float number is larger than `0.0`
- [x] `Float#negative?` Returns `true` if float number is less than `0.0`
- [x] `Float#zero?` Returns `true` if float number is equal to `0.0`

### CODE REVIEW

File: vm/float.go [hunk 1]
```diff
@@ -317,7 +317,239 @@ var builtinFloatInstanceMethods = []*BuiltinMethodObject{
...
+		// Returns the largest Integer less than or equal to self.
+		//
+		// ```Ruby
+		// 1.2.floor  # => 1
+		// 2.0.floor  # => 2
+		// -1.2.floor # => -2
+		// -2.0.floor # => -2
+		// ```
+		// @return [Integer]
+		Name: "floor",
```
Comment (saveriomiroddi) [inline vm/float.go:437]: It seems this is duplicated?

File: vm/float.go [hunk 2]
```diff
@@ -317,7 +317,239 @@ var builtinFloatInstanceMethods = []*BuiltinMethodObject{
...
+		// Returns the smallest Integer greater than or equal to self.
+		//
+		// ```Ruby
+		// 1.2.ceil  # => 2
+		// 2.ceil    # => 2
+		// -1.2.ceil # => -1
+		// -2.ceil   # => -2
+		// ```
+		// @return [Integer]
+		Name: "ceil",
```
Comment (saveriomiroddi) [inline vm/float.go:414]: I think this is duplicated, too.

File: vm/float.go [hunk 3]
```diff
@@ -317,7 +317,239 @@ var builtinFloatInstanceMethods = []*BuiltinMethodObject{
...
+	},
+	{
+		// Returns the Float as a positive value.
+		//
+		// ```Ruby
+		// -34.56.abs # => 34.56
+		// 34.56.abs # => 34.56
+		// ```
+		// @return [Float]
+		Name: "abs",
```
Comment (saveriomiroddi) [inline vm/float.go:394]: Third duplicate :grimacing:

File: vm/float.go (hunk 1 above)
Comment (st0012) [reply vm/float.go:437]: rebasing 🤦

**Comment by SD10** [comment]
🎉 Stay safe and healthy my friends

**Pull request merged**